frontend/public/components/catalog/catalog-item-details.jsx

```diff
@@ -80,6 +91,7 @@ export class CatalogTileDetails extends React.Component {
                 <SectionHeading text="Description" />
                 {tileDescription && <p>{tileDescription}</p>}
                 {markdownDescription && <SyncMarkdownView content={markdownDescription} />}
+                {helmChartReadme && <SyncMarkdownView content={helmChartReadme} />}
```

Comment:
Can we not make the markdownDescription field optionally a promise? If it's a promise we can perform a callback and wait for it to load.

This way we can co-locate the fetching of the helm readme contents in `normalizeHelmCharts`.

We have to think that at some point we will want turn all of these catalog contributions into extensions. In fact we already have a static extension for dev-catalog that we should be using.

Reply:
I looked at the mentioned extension when i was doing the Helm Charts catalog story for 4.4. We cannot use the static extension we have right now because it's very specific to k8s resources. And there is no actual fetching of resources involved in it. Just the normalize function is provided by the extension.

Reply:
Updated the PR to send a function that returns a promise as `markdownDescription`.